agent/doctor/doctor.go

```diff
@@ -0,0 +1,50 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//      http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package doctor
+
+import (
+	"github.com/cihub/seelog"
+)
+
+type Doctor struct {
+	healthchecks []Healthcheck
+}
+
+func NewDoctor() (*Doctor, error) {
```

Comment:
Having a "New" function that returns a Doctor that doesn't do anything doesnt seem like best interface to me. Perhaps the NewDoctor function should automatically call AddHeakthcheck with all the available healthchecks? Personally this seems like a better experience to me than every caller having to manually add each healthcheck even when using the "New" convenience function.

Reply:
good idea -- I'll update to let it push all the Healthchecks in